Source/ACE/Entity/Player.cs

```diff
@@ -1763,33 +1794,106 @@ public void HandleActionWieldItem(Container container, uint itemId, uint locatio
                             HandlePickupItem(container, itemGuid, location, PropertyInstanceId.Wielder);
                             return;
                         }
+
                         // Ok do you have this - if not bail
-                        if (GetInventoryItem(itemGuid) == null) return;
+                        var item = GetInventoryItem(itemGuid);
+                        if (item == null)
+                            return;
+                        if ((EquipMask)location > EquipMask.RingLeft)
+                        {
+                            // We are going into a weapon, wand or shield slot.
+                            switch ((EquipMask)location)
+                            {
+                                case EquipMask.MissileWeapon:
+                                    {
+                                        childLocation = 2;
+                                        placementId = 3;
+                                        break;
+                                    }
+                                case EquipMask.Shield:
+                                    {
+                                        childLocation = 3;
+                                        placementId = 6;
+                                        break;
+                                    }
+                                default:
+                                    {
+                                        placementId = 1;
+                                        childLocation = 1;
+                                        break;
+                                    }
+                            }
+                            Children.Add(new EquippedItem(itemGuid.Full, (EquipMask)location));
```

Comment:
This is something I don't quite understand from looking at the live server - but I have some idea why it is the way it is.   Only wielded shields and weapons are selectable on both other players and on monsters.   In live, unlike other wielded items like armor, jewelry or clothes - these items and only these items had both a wielder id AND a position in the world.   I can only speculate that to make that selectable and visible to be cast upon, it had to be known to the landblock.   That does bring up this question.   You could only have two items a weapon / caster and optionally a shield.   Both of those items had a parent_id set which was the wielder.   Now all wielded items have wielder set but only those with a position that are selectable have the parent_id set.   I wonder if when we get positions updates or (are calculating them from physics) if we update the position of the message sender then update on the child objects the position = to the parent position.   They are always the same.   Not sure there, we will need to figure that out.